controllers/binding/snapshotenvironmentbinding_controller.go

```diff
@@ -102,6 +110,22 @@ func (r *Reconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Resu
 
 	adapter := NewAdapter(snapshotEnvironmentBinding, snapshot, environment, application, integrationTestScenario, logger, loader, r.Client, ctx)
 
+	// check for failed bindingconditions and attempt to clean up Ephemeral environments
+	// see STONEINTG-433
+	bindingConditionsFailed := meta.FindStatusCondition(snapshotEnvironmentBinding.Status.BindingConditions, BindingErrorOccurredStatusConditionType)
+	if bindingConditionsFailed.Status == metav1.ConditionTrue {
```

Comment:
I think that we shouldn't have the business logic directly within the controller - both for testing and logic separation reasons.
I propose moving this condition to the beginning of the `EnsureEphemeralEnvironmentsCleanedUp` within the adapter and handle it similarly to how it's done within the other similar functions, e.g. [here](https://github.com/redhat-appstudio/integration-service/blob/c3828f452f3d80f65806265ee80cf291f504272b/controllers/binding/snapshotenvironmentbinding_adapter.go#L65-L68).